CS 410 Assignment 3/src/ass4/ControllerNotePad.java

```diff
@@ -0,0 +1,222 @@
+        model.getUndo().addActionListener(this);
+        model.getUndo().setActionCommand("undo");
+        model.getOpen().addActionListener(this);
+        model.getOpen().setActionCommand("open");
+        
+        model.getReplace().addActionListener(this);
+        model.getReplace().setActionCommand("replace");
+        
+        model.getRecent().setMnemonic(KeyEvent.VK_S);
+
+        
+     //   recent.addMenuListener(new MenuListener());
+
+        model.getRecent().addActionListener(this);
+        model.getRecent().setActionCommand("recent");
+        
+        model.getMenuBar().add(model.getFileMenu());
+        model.getMenuBar().add( model.getEditMenu());
+        setJMenuBar(model.getMenuBar());
+        add(new JScrollPane(model.getTextArea()));
+        setPreferredSize(new Dimension(600,600));
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setVisible(true);
+        pack();
+    
+    }
+    @Override
+    public void actionPerformed(ActionEvent e) {
+    	    	
+        if(e.getActionCommand().equals("new")) {
+        	model.getTextArea().setText("");
+        }else if(e.getActionCommand().equals("save")) {
+            File fileToWrite = null;
+            JFileChooser fc = new JFileChooser();
+            int returnVal = fc.showSaveDialog(null);
+            if (returnVal == JFileChooser.APPROVE_OPTION)
+                fileToWrite = fc.getSelectedFile();
+            try {
+                PrintWriter out = new PrintWriter(new FileWriter(fileToWrite));
+                out.println(model.getTextArea().getText());
+                JOptionPane.showMessageDialog(null, "File is saved successfully...");
+                out.close();
+            } catch (IOException ex) {
+            }
+        }else if(e.getActionCommand().equals("print")) {
+            try{
+                PrinterJob pjob = PrinterJob.getPrinterJob();
+                pjob.setJobName("Sample Command Pattern");
+                pjob.setCopies(1);
+                pjob.setPrintable(new Printable() {
+                    public int print(Graphics pg, PageFormat pf, int pageNum) {
+                        if (pageNum>0)
+                            return Printable.NO_SUCH_PAGE;
+                        pg.drawString(model.getTextArea().getText(), 500, 500);
+                        paint(pg);
+                        return Printable.PAGE_EXISTS;
+                    }
+                }
+                
+                		);
+
+                if (pjob.printDialog() == false)
+                    return;
+                pjob.print();
+            } catch (PrinterException pe) {
+                JOptionPane.showMessageDialog(null,
+                        "Printer error" + pe, "Printing error",
+                        JOptionPane.ERROR_MESSAGE);
+            }
+        }else if(e.getActionCommand().equals("copy")) {
+        	model.getTextArea().copy();
+        }else if(e.getActionCommand().equals("paste")) {

```

Comment:
So the first three lines of this "paste" block are unnecessary (dead code) because all you need to perform the paste function is:
model.getTextArea().paste();